### PR TITLE
Games

### PR DESCRIPTION
**Pull request by gonziesc**
travis
## Summary
- Create game feature
- Validated authorization
- Tests

## Trello Card
https://trello.com/c/9nVsz7vH/19-juego

### CODE REVIEW

File: app/controllers/games.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const gamesService = require('../services/games'),
+  session = require('../services/sessionManager'),
+  errors = require('../errors');
+
+exports.create = (req, res, next) => {
+  const game = req.body ? {
+    name: req.body.name,
+    code: req.body.code,
+    score: req.body.score
```
Comment (mishuagopian) [inline app/controllers/games.js]: Why score?

File: app/controllers/games.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  } : {};
+  const errorsSend = gamesService.validateGame(game);
+  if (errorsSend.length) {
+    next(errors.invalidCreate(errorsSend));
+  } else {
+    gamesService.create(game).then((g) => {
```
Comment (mishuagopian) [inline app/controllers/games.js]: remove unused g

File: app/models/game.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    name: {
+      type: Sequelize.STRING,
+      allowNull: false
+    },
+    code: {
+      type: Sequelize.INTEGER,
+      allowNull: false,
+      unique: true
+    },
+    score: {
```
Comment (mishuagopian) [inline app/models/game.js]: Why score ?

File: app/routes.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
 const auth = require('./middlewares/auth'),
   permisses = require('./middlewares/permisses'),
+  games = require('./controllers/games'),
   users = require('./controllers/users');
 
 exports.init = (app) => {
   app.get('/users', [auth.secure], users.getAll);
+  app.post('/games', [auth.secure, permisses.secure], games.create);
```
Comment (mishuagopian) [inline app/routes.js]: Remember, only for admins

File: test/games.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+const chai = require('chai'),
+  session = require('../app/services/sessionManager'),
+  server = require('./../app'),
+  should = chai.should();
+
+const successfulLogin = (cb) => {
+  chai.request(server)
```
Comment (mishuagopian) [inline test/games.js]: You can return this as a promise and use then in each `it` statement, without callbacks

File: app/controllers/games.js (hunk 1 above)
Comment (gonziesc) [inline app/controllers/games.js]: "Crear el modelo que representará a un juego. Los campos requeridos son: nombre, código y puntaje por acierto."

Comment (mishuagopian) [inline app/controllers/games.js]: 👌

File: app/controllers/games.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      res.end();
+    }).catch(next);
+  }
+};
+
+exports.getAll = (req, res) => {
+  gamesService.getAll(req.query.page, req.query.limit).then((games) => {
+    res.status(200);
+    res.send(games);
+  });
```
Comment (mishuagopian) [inline app/controllers/games.js]: catch missing here

**Comment by gonziesc** [comment]
Those html and css files are from htmls training. It was on this repo also, they were on master when i merged the branch.

**Pull request merged**